include/motor-control/core/brushed_motor/brushed_motor_interrupt_handler.hpp

```diff
@@ -213,6 +225,35 @@ class BrushedMotorInterruptHandler {
         return hardware.check_limit_switch();
     }
 
+    auto estop_triggered() -> bool { return hardware.check_estop_in(); }
+
+    void cancel_and_clear_moves() {
+        // If there is a currently running move send a error corrisponding
+        // to it so the hardware controller can know what move was running
+        // when the cancel happened
+        if (motor_state == ControlState::ACTIVE) {
+            static_cast<void>(
+                status_queue_client.send_brushed_move_status_reporter_queue(
+                    can::messages::ErrorMessage{
+                        .message_index = buffered_move.message_index,
+                        .severity = can::ids::ErrorSeverity::unrecoverable,
+                        .error_code = can::ids::ErrorCode::hardware}));
+        }
+        // send out a async error regardless of running a move
+        static_cast<void>(
```

Comment:
Does it mean we'll get two errors to the move status reporter queue if with different `message_indices` during a move?

Reply:
What's the purpose of having two error messages back to back?